File: week 6/readability.py
# TODO
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculate avg

    l = (letters / words) * 100
    s = (sentences / words) * 100
    index = round(0.0588 * l - 0.296 * s - 15.8)

    # Check grade

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

# Count letters


def count_letters(text):
    n = 0
    for i in text:
        if i.isalpha():
            n += 1
    print(n)
    return n

# Count words


def count_words(text):
    print(text.count(" ") + 1)
    return text.count(" ") + 1

# Count sentences


def count_sentences(text):
    sentences = 0
    sentences += text.count(".")
    sentences += text.count("?")
    sentences += text.count("!")
    print(sentences)
    return sentences


main()

File: week 2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    //turn counts to floats
    float letters = (float) count_letters(text);
    float words = (float) count_words(text);
    float sentences = (float) count_sentences(text);
    float L = letters / words * 100;
    float S = sentences / words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    //print grades
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

//iterate through letters
int count_letters(string text)
{
    int l = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //count if letter
        if (isalpha(text[i]))
        {
            l += 1;
        }
    }
    return l;
}

//iterate through words
int count_words(string text)
{
    int w = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //count if space
        if (isspace(text[i]))
        {
            w += 1;
        }
    }
    return w + 1;
}

//iterate through sentences
int count_sentences(string text)
{
    int s = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //count if sentence
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            s += 1;
        }
    }
    return s;
}